src/TagDependencyTrait.php

```diff
@@ -161,22 +186,21 @@ public static function loadModel(
      * Invalidate model tags.
      * @return bool
      */
-    public function invalidateTags()
+    public function invalidateTags($event)
     {
-
         /** @var TagDependencyTrait $this */
         \yii\caching\TagDependency::invalidate(
             $this->getTagDependencyCacheComponent(),
             [
                 static::commonTag(),
-                $this->objectTag()
+                $this->objectTag($event->name == ActiveRecord::EVENT_AFTER_UPDATE ? $event->changedAttributes : [])
             ]
         );
 
         if (!empty($this->cacheCompositeTagFields())) {
             \yii\caching\TagDependency::invalidate(
                 $this->getTagDependencyCacheComponent(),
-                $this->objectCompositeTag()
+                $this->objectCompositeTag($event->name == ActiveRecord::EVENT_AFTER_UPDATE ? $event->changedAttributes : [])
```

Comment:
I didn't think for a second my changes were best practices or anything, but my bad pasting skills aside, it's a possible solution, which actually keeps the cache synced in most cases without manually invalidating tags.